pkg/controller/common/util.go

```diff
@@ -1820,25 +1823,63 @@ func GetSnapshotClassForSmartClone(dvName string, targetPvcStorageClassName *str
 		return "", nil
 	}
 
-	// List the snapshot classes
-	scs := &snapshotv1.VolumeSnapshotClassList{}
-	if err := client.List(context.TODO(), scs); err != nil {
-		logger.Info("Cannot list snapshot classes, falling back to host assisted clone")
+	vscName, err := GetVolumeSnapshotClass(context.TODO(), client, targetStorageClass.Provisioner, snapshotClassName, logger)
+	if err != nil {
 		return "", err
 	}
-	for _, snapshotClass := range scs.Items {
-		// Validate association between snapshot class and storage class
-		if snapshotClass.Driver == targetStorageClass.Provisioner {
-			logger.Info("smart-clone is applicable for datavolume", "datavolume",
-				dvName, "snapshot class", snapshotClass.Name)
-			return snapshotClass.Name, nil
-		}
+	if vscName != nil {
+		logger.Info("smart-clone is applicable for datavolume", "datavolume",
+			dvName, "snapshot class", *vscName)
+		return *vscName, nil
 	}
 
 	logger.Info("Could not match snapshotter with storage class, falling back to host assisted clone")
 	return "", nil
 }
 
+// GetVolumeSnapshotClass looks up the snapshot class based on the driver and an optional specific name
+func GetVolumeSnapshotClass(ctx context.Context, c client.Client, driver string, snapshotClassName *string, log logr.Logger) (*string, error) {
+	logger := log.WithName("GetVolumeSnapshotClass").V(3)
+	if snapshotClassName != nil {
+		vsc := &snapshotv1.VolumeSnapshotClass{}
+		if err := c.Get(context.TODO(), types.NamespacedName{Name: *snapshotClassName}, vsc); err != nil {
+			return nil, err
+		}
+		if vsc.Driver == driver {
+			logger.Info("VolumeSnapshotClass selected according to StorageProfile", "name", vsc.Name)
```

Comment:
Choosing a specific VolumeSnapshotClass is a simple good-path logic and a rare use case, so I think logging it is more than enough (unlike the fallback to host-assisted clone where we added pvc event and anotation). @aglitke wdyt?

Reply:
Since this decision was made based on something other than the PVC/DV spec, I think we should err on the side of over-communicating this decision.  I agree with @awels .

Reply:
I guess the problem here is that we don't have the PVC so it is difficult to communicate this information.

Reply:
But we could pass in the name of the PVC, and use that to then look up the PVC and emit the event.